src/main/java/de/tum/cit/aet/artemis/core/web/PasskeyResource.java

```diff
@@ -0,0 +1,101 @@
+package de.tum.cit.aet.artemis.core.web;
+
+import static de.tum.cit.aet.artemis.core.config.Constants.PROFILE_CORE;
+
+import java.util.List;
+
+import jakarta.ws.rs.NotAllowedException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.web.webauthn.api.Bytes;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import de.tum.cit.aet.artemis.core.config.Constants;
+import de.tum.cit.aet.artemis.core.domain.User;
+import de.tum.cit.aet.artemis.core.domain.converter.BytesConverter;
+import de.tum.cit.aet.artemis.core.dto.PasskeyDto;
+import de.tum.cit.aet.artemis.core.repository.UserRepository;
+import de.tum.cit.aet.artemis.core.repository.webauthn.ArtemisUserCredentialRepository;
+import de.tum.cit.aet.artemis.core.security.annotations.EnforceAtLeastStudent;
+
+/**
+ * REST controller for public endpoints regarding the webauthn (Web Authentication) API, e.g. used for passkeys.
+ */
+@Profile(PROFILE_CORE)
+@RestController
+@RequestMapping("api/core/passkey/")
+public class PasskeyResource {
+
+    public static final String ENTITY_NAME = "passkey";
+
+    private static final Logger log = LoggerFactory.getLogger(PasskeyResource.class);
+
+    private final UserRepository userRepository;
+
+    private final ArtemisUserCredentialRepository artemisUserCredentialRepository;
+
+    @Value("${artemis.user-management.passkey.enabled:false}")
+    public boolean enabled;
+
+    /**
+     * REST controller for managing passkeys.
+     * Provides endpoints for retrieving and deleting passkeys associated with a user.
+     *
+     * This controller is only active when the "core" profile is enabled.
```

Comment:
This is a JavaDoc that would be fitting on the class level, not on the constructor level.